README.md

```diff
@@ -1703,3 +1711,100 @@ will return this response:
     }
 }
 ```
+
+## HTTP Server Proxy
+
+The proxy different stores can be used to both store and read their stored requests
+
+### Write to proxy store
+
+Whatever request performed against the server path `/proxy/<store_name>`.
+Where `<store_name>` are the keys inside the `proxy` object in the server configuration.
+The expected response will have `200` status code and no body.
+
+Given this request:
+```yaml
+{
+    "endpoint": "/proxy/test",
+    "method": "POST",
+    "query_params": {
+        "some": "param"
+    },
+    "header": {
+        "X-My-Header": 0
+    },
+    "body": {
+        "post": {
+            "my": ["body", "here"]
+        }
+    }
+}
+```
+
+The expected response: 
+```yaml
+{
+    "statuscode": 200
+}
+```
+
+### Read from proxy store
+
+Whatever request performed against the server path `/proxystore/<store_name>?offset=<offset>&limit=<limit>`.
+Where:
+- `<store_name>` is a key inside the `proxy` object in the server configuration, aka the proxy store name
+- `<offset>` represents the first entry to be retrieved in the proxy store requests collection.
+- `<limit>` represents the maximum amount of entries to be retrieved from the proxy store.
+
+Given this request:
+```yaml
+{
+    "endpoint": "/proxystore/test",
+    "method": "GET",
+    "query_params": {
+        "offset": 0,
+        "limit": 2
+    }
+}
+```
+
+The expected response: 
+```yaml
+{
+     "mode": "passthru", // The mode the proxy store runs on
+     "offset": 0, // Offset requested
+     "next_offset": 2, // Next offset, as the current offset + limit or 0 if reached max entries
+     "limit": 2, // Limit requested
+     "count": 20, // Total number of requests recorded in this proxy store
+     "store": [
+           {
+                "offset": 0, // The offset for this request entry
+                "request": {
+                       "method": "POST", // The method of this request to the proxy store
+                       "header": { // The headers of this request to the proxy store
+                           "X-My-Header": ["blah"]
+                       }
+                       "query": { // The query string parameters of this request to the proxy store
+                           "nolimit": [true]
+                       }
+                       "body": { // The body of this request to the proxy store
```

Comment:
Missing here is the "path" requested. May be a user uses their own format for the query string, so that the go query parser will not output anything. "path" is the full path as received by the proxy, including the query string.